safe/gis/expressions.py

```diff
@@ -0,0 +1,37 @@
+# coding=utf-8
+from qgis.core import (
+    qgsfunction,
+    QgsMapLayerRegistry,
+    QgsExpressionContextUtils,
+)
+from safe.definitions.provenance import provenance_layer_analysis_impacted_id
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+
+@qgsfunction(
+    args='auto', group='InaSAFE', usesGeometry=False, referencedColumns=[])
+def inasafe_impact_analysis_layer(field, feature, parent):
+    """Retrieve a value from a field in the impact analysis layer."""
+    _ = feature, parent  # NOQA
+    project_context_scope = QgsExpressionContextUtils.projectScope()
+    registry = QgsMapLayerRegistry.instance()
+
+    key = provenance_layer_analysis_impacted_id['provenance_key']
```

Comment:
Yes, definitely, we want to encourage users to create their own map template. That's why we are doing this expression system. Then can create their own QPT by using the powerful QGIS system expression.

I just made for the analysis layer.
What do you need for you? Just let me know which other layer you need.